src/services/ContributingDeveloperAuditService/scm/GitHub/api/GitHubApiClient.ts

```diff
@@ -0,0 +1,215 @@
+      },
+      (rejectedRequest) => {
+        return Promise.reject(rejectedRequest);
+      },
+    );
+
+    client.interceptors.response.use(
+      async (response) => {
+        soosLogger.verboseDebug(apiClientName, `Response Body: ${JSON.stringify(response.data)}`);
+        if (response.config.url?.includes("per_page")) {
+          return await GitHubApiClient.handlePagination(response, client);
+        }
+        return response;
+      },
+      async (error) => {
+        const { config, response } = error;
+        const maxRetries = 3;
+        config.retryCount = config.retryCount || 0;
+
+        if (response?.status === 429 && config.retryCount < maxRetries) {
+          config.retryCount += 1;
+          soosLogger.verboseDebug(
+            apiClientName,
+            `Retrying request (${config.retryCount}) after 3 minutes due to 429 status.`,
+          );
+          await sleep(180000);
+          return client(config);
+        }
+
+        if (response?.status) {
+          soosLogger.verboseDebug(apiClientName, `Response Status: ${response.status}`);
+        }
+        return Promise.reject(error);
+      },
+    );
+
+    return client;
+  }
+
+  private static async handlePagination(
+    response: AxiosResponse,
+    client: AxiosInstance,
+  ): Promise<AxiosResponse> {
+    let data = response.data;
+    let nextUrl = GitHubApiClient.getNextPageUrl(response);
+    while (nextUrl) {
+      soosLogger.verboseDebug("Handling pagination for ", response.config.url);
+      soosLogger.verboseDebug("Next url being fetched for pagination", nextUrl);
+      const nextPageResponse = await client.get(nextUrl);
+      data = data.concat(nextPageResponse.data);
+      nextUrl = GitHubApiClient.getNextPageUrl(nextPageResponse);
+    }
+
+    return { ...response, data };
+  }
+
+  private static getNextPageUrl(response: AxiosResponse): string | null {
+    const linkHeader = response.headers["link"] as string | undefined;
+    const nextLink = linkHeader?.split(",").find((s) => s.includes('rel="next"'));
+    return nextLink
+      ? new URL(nextLink.split(";")[0].trim().slice(1, -1), response.config.baseURL).toString()
+      : null;
+  }
+
+  async getGithubOrgs(): Promise<GitHubOrganization[]> {
+    const response = await this.client.get<GitHubOrganization[]>(`/user/orgs?per_page=100`);
+
+    const orgs: GitHubOrganization[] = response.data;
+    return orgs;
+  }
+
+  async getGitHubOrgRepos(org: GitHubOrganization): Promise<GitHubRepository[]> {
+    const response = await this.client.get<GitHubRepository[]>(
+      `/orgs/${org.login}/repos?per_page=50`,
+    );
+
+    const repos: GitHubRepository[] = response.data;
+    return repos;
+  }
+
+  async getContributorsForRepo(repository: GitHubRepository): Promise<ContributingDeveloper[]> {
+    const response = await this.client.get<Commits[]>(
+      `/repos/${repository.owner.login}/${repository.name}/commits?per_page=100&since=${threeMonthsDate}`,
```

Comment:
I'd rename the threeMonthsDate to "commitsSinceDate" or something more generic that doesn't refer to it's value.